src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -108,27 +109,27 @@ internal class LanguageWorkerChannel : ILanguageWorkerChannel, IDisposable
 
         internal ILanguageWorkerProcess WorkerProcess => _languageWorkerProcess;
 
-        public Task StartWorkerProcessAsync()
+        public async Task<Task> StartWorkerProcessAsync()
         {
             _startSubscription = _inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.StartStream)
                 .Timeout(TimeSpan.FromSeconds(LanguageWorkerConstants.ProcessStartTimeoutSeconds))
                 .Take(1)
-                .Subscribe(SendWorkerInitRequest, HandleWorkerChannelError);
-
-            _languageWorkerProcess.StartProcess();
+                .Subscribe(SendWorkerInitRequest, HandleWorkerStartStreamError);
 
+            _workerChannelLogger.LogDebug("Initiating Worker Process start up");
+            await _languageWorkerProcess.StartProcess();
             _state = LanguageWorkerChannelState.Initializing;
-
             return _workerInitTask.Task;
```

Comment:
Could we return Task<bool> and do `return await _workerInitTask.Task` instead?

Reply:
Yes! Thanks for pointing it out. Will update.